test_cases/stdlib/check_dataclasses.py

```diff
@@ -92,10 +92,10 @@ def check_other_isdataclass_overloads(x: type, y: object) -> None:
 
 # Regression test for #11653
 D = dc.make_dataclass(
-    "D", [("a", int | None), "y", ("z", Annotated[frozenset[bytes], "metadata"], dc.field(default=frozenset()))]
+    "D", [("a", Union[int, None]), "y", ("z", Annotated[FrozenSet[bytes], "metadata"], dc.field(default=frozenset()))]
 )
 # Check that it's inferred by the type checker as a class object of some kind
 # (but don't assert the exact type that `D` is inferred as,
 # in case a type checker decides to add some special-casing for
 # `make_dataclass` in the future)
-assert_type(D.__mro__, tuple[type, ...])
+_: type = D
```

Comment:
I almost posted a comment suggesting this, but note that it would pass even if the type checker infers that make_dataclass() returns Any.

Reply:
You did post it, and I applied your suggestion before you deleted it 😄

I'll revert it back to how I originally had it. Though pyright seems unhappy with something else for some reason.

Reply:
Ah you have less psychic powers than I thought :D